tensorflow/core/api_def/base_api/api_def_ScatterNd.pbtxt

```diff
@@ -30,33 +30,28 @@ END
   }
   summary: "Scatters `updates` into a tensor of shape `shape` according to `indices`."
   description: <<END
-Update the input tensor by scattering sparse `updates` according to individual values at the specified `indices`.
-This op returns an `output` tensor with the `shape` you specify. This op is the
-inverse of the `tf.gather_nd` operator which extracts values or slices from a
-given tensor.
+Scatter sparse `updates` according to individual values at the specified
+`indices`. This op returns an output tensor with the `shape` you specify. This
+op is the inverse of the `tf.gather_nd` operator which extracts values or slices
+from a given tensor.
 
 This operation is similar to `tf.tensor_scatter_nd_add`, except that the tensor
-is zero-initialized. Calling `tf.scatter_nd(indices, values, shape)`
+is zero-initialized. Calling `tf.scatter_nd(indices, updates, shape)`
 is identical to calling
-`tf.tensor_scatter_nd_add(tf.zeros(shape, values.dtype), indices, values)`
+`tf.tensor_scatter_nd_add(tf.zeros(shape, updates.dtype), indices, updates)`
 
-If `indices` contains duplicates, the duplicate `values` are accumulated
-(summed).
+If `indices` contains duplicates, the associated `updates` are accumulated
+(summed) into the output tensor.
 
-**WARNING**: The order in which updates are applied is nondeterministic, so the
-output will be nondeterministic if `indices` contains duplicates;
-numbers summed in different order may yield different results because of some
-numerical approximation issues.
-
-`indices` is an integer tensor of shape `shape`. The last dimension
-of `indices` can be at most the rank of `shape`:
+`indices` is an integer tensor containing indices into the output tensor. The
+last dimension of `indices` can be at most the rank of `shape`:
 
     indices.shape[-1] <= shape.rank
 
 The last dimension of `indices` corresponds to indices of elements
 (if `indices.shape[-1] = shape.rank`) or slices
-(if `indices.shape[-1] < shape.rank`) along dimension `indices.shape[-1]` of
-`shape`.
+(if `indices.shape[-1] < shape.rank`) along dimension `indices.shape[-1]`
+(counting from 1) of `shape`.
```

Comment:
I don't think "(counting from 1)" is accurate, since it counts from 0. For example, suppose we have:

```
indices = [[2], [0]]  # shape=(2, 1)
updates = [[9, 8], [7, 6]]
shape = (3, 2)
tf.scatter_nd(indices, updates, shape)

# Output is
# [[7, 6],
#  [0, 0],
#  [9, 8]]
```

`indices.shape[-1]` is 1. The indices, [2] and [0], represent slices of size 2 along the last dimension of the output (the dimension with index 1).  The slices (of size 2) are replaced by the values [9, 8] and [7, 6] in `updates`.

Reply:
I agree with you. The following clarification helped me.

Let x = indices.shape[-1]
Output is sliced along dimension x-1 (see usage in `[tf.reverse_sequence](https://www.tensorflow.org/api_docs/python/tf/reverse_sequence)`) producing "slices along dimension x" and higher dimensions counting from zero (see usage in [Wikipedia > Array Slicing](https://en.wikipedia.org/wiki/Array_slicing#1970s:_MATLAB) counting from 1).